core/src/main/java/io/substrait/expression/ExpressionCreator.java

```diff
@@ -102,17 +114,59 @@ public static Expression.TimestampLiteral timestamp(
             .withNano((int) TimeUnit.MICROSECONDS.toNanos(micros)));
   }
 
+  /**
+   * @deprecated TimestampTZ is deprecated in favor of PrecisionTimestampTZ
+   */
+  @Deprecated
   public static Expression.TimestampTZLiteral timestampTZ(boolean nullable, long value) {
     return Expression.TimestampTZLiteral.builder().nullable(nullable).value(value).build();
   }
 
+  /**
+   * @deprecated TimestampTZ is deprecated in favor of PrecisionTimestampTZ
+   */
+  @Deprecated
   public static Expression.TimestampTZLiteral timestampTZ(boolean nullable, Instant value) {
     var epochMicro =
         TimeUnit.SECONDS.toMicros(value.getEpochSecond())
             + TimeUnit.NANOSECONDS.toMicros(value.getNano());
     return timestampTZ(nullable, epochMicro);
   }
 
+  public static Expression.PrecisionTimestampLiteral precisionTimestamp(
+      boolean nullable, long value, int precision) {
+    return Expression.PrecisionTimestampLiteral.builder()
+        .nullable(nullable)
+        .value(value)
+        .precision(precision)
+        .build();
+  }
+
+  public static Expression.PrecisionTimestampLiteral precisionTimestamp(
+      boolean nullable, LocalDateTime value) {
+    var epochMicro =
+        TimeUnit.SECONDS.toMicros(value.toEpochSecond(ZoneOffset.UTC))
+            + TimeUnit.NANOSECONDS.toMicros(value.toLocalTime().getNano());
+    return precisionTimestamp(nullable, epochMicro, 6);
+  }
+
+  public static Expression.PrecisionTimestampTZLiteral precisionTimestampTZ(
+      boolean nullable, long value, int precision) {
+    return Expression.PrecisionTimestampTZLiteral.builder()
+        .nullable(nullable)
+        .value(value)
+        .precision(precision)
+        .build();
+  }
+
+  public static Expression.PrecisionTimestampTZLiteral precisionTimestampTZ(
+      boolean nullable, Instant value) {
+    var epochMicro =
+        TimeUnit.SECONDS.toMicros(value.getEpochSecond())
+            + TimeUnit.NANOSECONDS.toMicros(value.getNano());
+    return precisionTimestampTZ(nullable, epochMicro, 6);
```

Comment:
The docs for [Instant](https://docs.oracle.com/javase/8/docs/api/java/time/Instant.html) indicate that
> the class stores a long representing epoch-seconds and an int representing nanosecond-of-second

Should we be using nanoseconds (9) here instead of microseconds (6)?

Relatedly, are all possible Instant values representable as PrecistionTimestampTz literals?

Reply:
See the other answer. Not all instants are representable as PTTz, I believe, since instant stores in total 96 bits of data, while PTTz is only 64 bits.